catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/filter/AbstractFilterVisitor2.java

```diff
@@ -54,163 +38,110 @@
+    traceName(visitable);
+    visitObjects(visitable);
   }
 
   @Override
-  public void visitDistanceBufferType(VisitableXmlElement<DistanceBufferType> visitable) {
-    JAXBElement<DistanceBufferType> element = visitable.getElement();
-    String localPart = element.getName().getLocalPart();
-    traceLocalPart(localPart);
+  public void visitBinarySpatialType(VisitableElement<List<Object>> visitable) {
+    traceName(visitable);
+    visitObjects(visitable);
   }
 
   @Override
-  public void visitBoundingBoxType(VisitableXmlElement<BBOXType> visitable) {
-    JAXBElement<BBOXType> element = visitable.getElement();
-    String localPart = element.getName().getLocalPart();
-    traceLocalPart(localPart);
+  public void visitDistanceBufferType(VisitableElement<List<Object>> visitable) {
+    traceName(visitable);
+    throw new UnsupportedOperationException("DistanceBufferType currently is not supported");
   }
 
   @Override
-  public void visitBinaryLogicType(VisitableXmlElement<BinaryLogicOpType> visitable) {
-    JAXBElement<BinaryLogicOpType> element = visitable.getElement();
-    String localPart = element.getName().getLocalPart();
-    traceLocalPart(localPart);
-
-    element.getValue().getOps().forEach(jax -> makeVisitable(jax).accept(this));
+  public void visitBoundingBoxType(VisitableElement<List<Object>> visitable) {
+    traceName(visitable);
+    throw new UnsupportedOperationException("BoundingBoxType currently is not supported");
   }
 
   @Override
-  public void visitBinaryComparisonType(VisitableXmlElement<BinaryComparisonOpType> visitable) {
-    JAXBElement<BinaryComparisonOpType> element = visitable.getElement();
-    String localPart = element.getName().getLocalPart();
-    traceLocalPart(localPart);
-
-    element.getValue().getExpression().forEach(jax -> makeVisitable(jax).accept(this));
+  public void visitBinaryComparisonType(VisitableElement<List<VisitableElement<?>>> visitable) {
+    traceName(visitable);
+    visitable.getValue().forEach(v -> v.accept(this));
   }
 
   @Override
-  public void visitPropertyIsLikeType(VisitableXmlElement<PropertyIsLikeType> visitable) {
-    JAXBElement<PropertyIsLikeType> element = visitable.getElement();
-    String localPart = element.getName().getLocalPart();
-    traceLocalPart(localPart);
+  public void visitPropertyIsLikeType(VisitableElement<List<VisitableElement<?>>> visitable) {
+    traceName(visitable);
+    visitable.getValue().forEach(v -> v.accept(this));
   }
 
   @Override
-  public void visitPropertyIsNullType(VisitableXmlElement<PropertyIsNullType> visitable) {
-    JAXBElement<PropertyIsNullType> element = visitable.getElement();
-    String localPart = element.getName().getLocalPart();
-    traceLocalPart(localPart);
+  public void visitPropertyIsNullType(VisitableElement<VisitableElement<?>> visitable) {
+    traceName(visitable);
+    throw new UnsupportedOperationException("PropertyIsNullType currently is not supported");
```

Comment:
Should we make a TODO and ticket for the unsupported operations?

Reply:
https://codice.atlassian.net/browse/DDF-3829